Source/WebCore/page/scrolling/AsyncScrollingCoordinator.cpp

```diff
@@ -107,6 +106,40 @@ void AsyncScrollingCoordinator::handleWheelEventPhase(ScrollingNodeID nodeID, Pl
 }
 #endif
 
+ScrollingStateTree& AsyncScrollingCoordinator::ensureScrollingStateTreeForRootFrameID(FrameIdentifier identifier)
+{
+    auto it = m_scrollingStateTrees.find(identifier);
+    if (it != m_scrollingStateTrees.end())
+        return it->value.get();
+
+    ASSERT_WITH_MESSAGE(!page() || page()->settings().siteIsolationEnabled() || m_scrollingStateTrees.isEmpty(), "With site isolation off, we should only ever have one scoling state tree. Get the frame identifier from LocalFrame::rootFrame.");
+    auto addResult = m_scrollingStateTrees.set(identifier, makeUniqueRef<ScrollingStateTree>(this));
+    return addResult.iterator->value.get();
+}
+
+void AsyncScrollingCoordinator::removeRootFrame(FrameIdentifier identifier)
+{
+    m_scrollingStateTrees.remove(identifier);
+}
+
+RefPtr<ScrollingStateNode> AsyncScrollingCoordinator::stateNodeForID(ScrollingNodeID nodeID) const
+{
+    for (auto& tree : m_scrollingStateTrees.values()) {
+        if (RefPtr scrollingNode = tree->stateNodeForID(nodeID))
+            return scrollingNode;
+    }
+    return nullptr;
+}
+
+ScrollingStateTree* AsyncScrollingCoordinator::stateTreeForID(ScrollingNodeID nodeID) const
```

Comment:
`stateTreeForNodeID()` to disambiguate from other kinds of IDs.

Reply:
Fixed.